app/code/Magento/SalesSequence/Model/ResourceModel/Profile.php

```diff
@@ -77,4 +77,21 @@ public function loadActiveProfile($metadataId)
         }
         return $profile;
     }
+
+    /*
+     * Get profile ids by metadata id
+     * @param int $metadataId
+     * @return int[]
+     */
+    public function getProfileIds($metadataId)
+    {
+        $connection = $this->getConnection();
+        $bind = ['meta_id' => $metadataId];
+        $select = $connection->select()
+            ->from($this->getMainTable(), ['profile_id'])
+            ->where('meta_id = :meta_id')
+            ->where('is_active = 1');
```

Comment:
I think that non-active profiles should be removed as well in case of store delete